### PR TITLE
Make the app navigation (slot) a real component and add a spacer

### PR DESCRIPTION
**Pull request by ChristophWurst**
This makes it a lot easier for apps to create their own reusable navigation component.

Without this change, using an own component to make the navigation slot contents reusable resulted in an extra element wrapping the contents. This can easily break the design. I think the clean version is to just have a dedicated component as we have for many other things now.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
See https://github.com/nextcloud/mail/pull/1576 for how I used the code in an app.

File: src/components/AppNavigationSpacer/AppNavigationSpacer.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+</template>
+
+<script>
+export default {
+	name: 'AppNavigationSpacer'
+}
+</script>
+
+<style scoped>
+	li {
```
Comment (korelstar) [inline src/components/AppNavigationSpacer/AppNavigationSpacer.vue]: Elements selectors should be avoided with `scoped`: https://vuejs.org/v2/style-guide/#Element-selectors-with-scoped-use-with-caution

Comment (ChristophWurst) [reply src/components/AppNavigationSpacer/AppNavigationSpacer.vue]: Thanks! I didn't know that :)

File: src/components/AppNavigationSpacer/AppNavigationSpacer.vue [hunk 2]
```diff
@@ -20,7 +20,7 @@
  -
  -->
 <template>
-	<li />
+	<li class="spacer" />
```
Comment (skjnldsv) [inline src/components/AppNavigationSpacer/AppNavigationSpacer.vue]: Let's keep the same class structure? :see_no_evil: 
`app-navigation-spacer` ?

Comment (ChristophWurst) [reply src/components/AppNavigationSpacer/AppNavigationSpacer.vue]: sure!

File: src/components/AppNavigation/AppNavigation.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	<div id="app-navigation" :class="navigationClass">
+		<slot />
+	</div>
+</template>
+
+<script>
+export default {
+	name: 'AppNavigation',
+	props: {
+		navigationClass: {
```
Comment (korelstar) [inline src/components/AppNavigation/AppNavigation.vue]: I think the property `navigationClass` is not needed anymore due to moving this element to it's own component. You can use `:class`, directly:

```
<app-navigation :class="{loading: loading}"> ... </app-navigation>
```

Comment (skjnldsv) [reply src/components/AppNavigation/AppNavigation.vue]: Ah yes! You're right :see_no_evil: 
@ChristophWurst

Comment (ChristophWurst) [reply src/components/AppNavigation/AppNavigation.vue]: Of course :see_no_evil: 

@korelstar could you submit a PR please? :)

Comment (korelstar) [reply src/components/AppNavigation/AppNavigation.vue]: Yep, please see #296

**Pull request merged**